src/compiler/core.ts

```diff
@@ -286,6 +285,7 @@ export function filter<T>(array: T[], f: (x: T) => boolean): T[];
 /** @internal */
 export function filter<T, U extends T>(array: readonly T[], f: (x: T) => x is U): readonly U[];
 /** @internal */
+// eslint-disable-next-line @typescript-eslint/no-unused-vars

```

Comment:
I think we should be able to drop the type vars in this file.

Reply:
The reason why all these are suppressed is because TypeScript requires that all overloads of a function must accept the same type parameters. So while the type parameter may be unused for a given overload, removing it causes TS errors.

Reply:
I don't think that's true; see also my PR https://github.com/microsoft/TypeScript/pull/55139/files#diff-8eee8edc4cdf82aaa3b0c0ba3c8c74fa80ac6d5214540a460c49fb2ee1031b25R289